stable_baselines3/td3/td3.py

```diff
@@ -158,6 +158,7 @@ def train(self, gradient_steps: int, batch_size: int = 100) -> None:
                 target_q_values = replay_data.rewards + (1 - replay_data.dones) * self.gamma * next_q_values
 
             # Get current Q-values estimates for each critic network
+            self.critic.set_training_mode(True)
```

Comment:
dito

Reply:
The reason for this is similar to the case above. 

On lines 175-177 we have 

```
# Compute actor loss
self.critic.set_training_mode(False)
actor_loss = -self.critic.q1_forward(replay_data.observations, self.actor(replay_data.observations)).mean()
```

So what I'm doing is putting `self.critic` into eval mode when updating `self.actor`, as `self.critic` is not updated at this point. Then I have to put `self.critic` back into train mode on line 161 before it's updated on the next iteration.

Again, I _think_ that this is the correct behaviour, but it would be good to hear your opinion.

Reply:
I see, that was mostly to keep the `train()` method clean

Reply:
To be honest, I'm not really sure it will make much difference to the training process. I can remove these calls if you prefer to keep the code clean?

Reply:
Yes, please do. (if we remove it, it would only slightly update the batch norm stats which should be fine)
Also, it will allow the dropout to be active there.

Reply:
Cool. I've removed the extra calls to `set_training_mode`.